data-model/src/main/java/io/micronaut/data/model/naming/NamingStrategy.java

```diff
@@ -119,16 +119,24 @@ public interface NamingStrategy {
             Association owningAssociation = inverseSide.orElse(association);
             return mappedName(owningAssociation.getOwner().getDecapitalizedName() + owningAssociation.getAssociatedEntity().getSimpleName());
         } else {
-            switch (association.getKind()) {
-                case ONE_TO_ONE:
-                case MANY_TO_ONE:
-                    return mappedName(association.getName() + getForeignKeySuffix());
-                default:
-                    return mappedName(association.getName());
-            }
+            return switch (association.getKind()) {
+                case ONE_TO_ONE, MANY_TO_ONE -> mappedName(association.getName() + getForeignKeySuffix());
+                default -> mappedName(association.getName());
+            };
         }
     }
 
+    /**
+     * Convert the associated name to a proper format to be appended to the path.
+     * @param associatedName The associated name
+     * @return the name in a proper format
+     * @since 4.1.0
```

Comment:
Will this be 4.2.0 (the latest is 4.1.2)?

Reply:
Fixed